src/components/UI/alert.tsx

```diff
@@ -0,0 +1,150 @@
+  content: AlertContentProps
+}
+
+const Alert: React.FC<AlertProps> = ({ visible, setVisible, content }) => {
+  const [contentCache, setContentCache] = useState<AlertContentProps>({
+    title: '', // string[2], first element is tile, second is message
+    text: '',
+    position: 'top',
+    confirmFunction: () => {
+      return
+    }, // function to execute on confirm, enables the confirm function
+    cancelFunction: () => {
+      return
+    },
+    showFor: 5000, // ms alert will stay open, set to -1 to leave open until button click
+    variant: AlertVariant.info
+  })
+
+  const timerRef = useRef<ReturnType<typeof setTimeout>>()
+
+  // utility function that sets and then clears a timer to automatically close alert
+  const doUpdate = useCallback(
+    (c: AlertContentProps) => {
+      if (c.text !== '') {
+        setContentCache({
+          title: c.title,
+          text: c.text,
+          position: c.position,
+          confirmFunction: c.confirmFunction, // function to execute on confirm, enables the confirm function
+          cancelFunction: c.cancelFunction,
+          showFor: c.showFor, // ms alert will stay open, set to -1 to leave open until button click
+          variant: c.variant
+        })
+        setVisible(true)
+        if (c.showFor !== -1) {
+          timerRef.current = setTimeout(() => {
+            setVisible(false)
+          }, content.showFor)
+        }
+      }
+    },
+    [content, setVisible]
+  )
+
+  useEffect(() => {
+    if (timerRef.current) clearTimeout(timerRef.current)
+    doUpdate(content)
+  }, [content, doUpdate])
+
+  useEffect(() => {
+    return () => {
+      if (timerRef.current) clearTimeout(timerRef.current)
+    }
+  }, [])
+
+  return (
+    <div
+      className={getContainerClasses(
+        visible,
+        contentCache.position ? contentCache.position : 'top'
+      )}
+      style={{ boxShadow: shadow }}
+    >
+      <div className='h-7 w-7'>
+        {contentCache.variant === AlertVariant.info ? (
+          <InformationCircleIcon className='h-7 w-7' />
+        ) : contentCache.variant === AlertVariant.security ? (
+          <ShieldExclamationIcon className='h-7 w-7' />
+        ) : contentCache.variant === AlertVariant.critical ? (
+          <ExclamationIcon className='h-7 w-7' />
+        ) : (
+          contentCache.variant === AlertVariant.comment && (
+            <AnnotationIcon className='h-7 w-7' />
+          )
+        )}
+      </div>
+      <div className='mx-[1rem] grow self-start'>
+        <p className='font-bold' style={{ color: titleColor }}>
+          {contentCache.title}
+        </p>
+        <p style={{ color: textColor }}>{contentCache.text}</p>
+      </div>
+      {typeof contentCache.confirmFunction !== 'undefined' && (
+        <button
+          className='mr-4'
+          onClick={(e) => {
+            e.preventDefault()
+            setVisible(false)
+            if (typeof contentCache.confirmFunction !== 'undefined')
```

Comment:
I thought the same but the first includes the checkmark button, the second is within the function constant called at the onclick event, so even though it may be redundant it makes sense to me at the onclick to make sure the function is not undefined.